executor/ddl.go

```diff
@@ -438,12 +515,41 @@ func (e *DDLExec) dropTableObject(objects []*ast.TableName, obt objectType, ifEx
 	if len(notExistTables) > 0 && ifExists {
 		for _, table := range notExistTables {
 			if obt == sequenceObject {
-				e.ctx.GetSessionVars().StmtCtx.AppendNote(infoschema.ErrSequenceDropExists.GenWithStackByArgs(table))
+				sessVars.StmtCtx.AppendNote(infoschema.ErrSequenceDropExists.GenWithStackByArgs(table))
 			} else {
-				e.ctx.GetSessionVars().StmtCtx.AppendNote(infoschema.ErrTableDropExists.GenWithStackByArgs(table))
+				sessVars.StmtCtx.AppendNote(infoschema.ErrTableDropExists.GenWithStackByArgs(table))
 			}
 		}
 	}
+
+	if obt == tableObject && len(localTempTables) > 0 {
+		err := e.dropLocalTemporaryTables(localTempTables)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (e *DDLExec) dropLocalTemporaryTables(localTempTables []*ast.TableName) error {
```

Comment:
You can change paramter `localTempTables []*ast.TableName` to `localTempTables []*model.TableInfo`, so you need not to get table info again.